backports_repo.py

```diff
@@ -63,51 +68,81 @@ def linkpac(pkg, proj_source):
     return ret
 
 
-backports_python = {x for x in project_list(src_URL.format(project))}
-factory_python = {x for x in project_list(src_URL.format(factory_name))
-                  if x.startswith('python')}
-python_itself = {"python-base", "python3-base", "python", "python3",
-                 "python-doc", "python3-doc"}
-
-additional_conf_file = os.path.join(appdirs.user_config_dir(),
-                                    'osc', 'backports_repo.json')
-
-# Fall back to in-tree copy
-if not os.path.exists(additional_conf_file):
-    additional_conf_file = os.path.join(
+def _get_additional_links_from_config():
+    conf_file = os.path.join(appdirs.user_config_dir(),
+                             'osc', 'backports_repo.json')
+    conf_file_fallback = os.path.join(
         os.path.dirname(os.path.abspath(__file__)), 'backports_repo.json')
-
-# extra packages we want there
-additional_links = set()
-if os.path.exists(additional_conf_file):
-    with io.open(additional_conf_file) as conf_f:
-        obj = json.load(conf_f)
-        if 'additional_links' not in obj:
-            raise KeyError(
-                '"additional_links" not found in configuration file %s' %
-                additional_conf_file)
-        additional_links = set(obj['additional_links'])
-
-factory_python = factory_python | additional_links
-factory_python -= {
-    'python-Django',  # prefer python-Django1
-}
-
-
-futures = []
-with concurrent.futures.ThreadPoolExecutor(max_workers=MAX_PROCS) as executor:
-    # remove packages not in tumbleweed
-    for package in backports_python - factory_python:
-        futures.append(executor.submit(rdelete, package, project))
-
-    # add packages not in yet
-    for package in factory_python - python_itself - backports_python:
-        futures.append(executor.submit(linkpac, package, project))
-
-results = []
-for f in concurrent.futures.as_completed(futures):
-    results.append(f.result())
-
-results = [x for x in results if x != 0]
-
-sys.exit(len(results))
+    for f in [conf_file, conf_file_fallback]:
+        if not os.path.exists(f):
```

Comment:
This changes behavior, now the fallback file is preferred when both are available

Reply:
No. The return in line 86 is inside of the loop. so the first available file with vaid data is used.

Reply:
ah, the return. indeed.